src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -1043,77 +1078,68 @@ protected virtual IDbDataParameter CreateDatabaseParameter(IDbCommand command, D
         /// <param name="parameterInfo">Parameter configuration info.</param>
         protected internal virtual object GetDatabaseParameterValue(LogEventInfo logEvent, DatabaseParameterInfo parameterInfo)
         {
-            Type dbParameterType = parameterInfo.ParameterType;
-            if (string.IsNullOrEmpty(parameterInfo.Format) && dbParameterType == typeof(string))
-            {
-                return RenderLogEvent(parameterInfo.Layout, logEvent) ?? string.Empty;
-            }
+            return RenderObjectValue(logEvent, parameterInfo.Name, parameterInfo.Layout, parameterInfo.ParameterType, parameterInfo.Format, parameterInfo.Culture);
+        }
 
-            IFormatProvider dbParameterCulture = GetDbParameterCulture(logEvent, parameterInfo);
+        private object GetDatabaseConnectionValue(LogEventInfo logEvent, DatabaseConnectionInfo connectionInfo)
+        {
+            return RenderObjectValue(logEvent, connectionInfo.Name, connectionInfo.Layout, connectionInfo.PropertyType, connectionInfo.Format, connectionInfo.Culture);
```

Comment:
proposal: create an interface for DatabaseParameterInfo and DatabaseConnectionProperty, and pass that interface to RenderObjectValue.

Reply:
We need to add some internal readonly properties for the common interface on DatabaseParameterInfo and DatabaseConnectionProperty then, e.g.ValueType and ValueLayout

Reply:
e.g. 


```c#
internal interface IDatabaseRenderable
{
    string Name { get; }
    Layout Layout { get; }
    Type ValueType { get; }
    string Format { get; }
    CultureInfo Culture { get; }
}

```

Reply:
Sounds like an mental exercise for you.

Reply:
Yeah if you OK with this, I could push it to this branch

Reply:
Sure push away. Have committed the code-review changes agreed upon.

Reply:
OK unresolve as reminder